packages/webapp/src/components/Animals/AnimalSingleViewHeader/index.tsx

```diff
@@ -0,0 +1,206 @@
+import { getAge } from '../../../util/age';
+import {
+  Animal,
+  AnimalBatch,
+  CustomAnimalBreed,
+  CustomAnimalType,
+  DefaultAnimalBreed,
+  DefaultAnimalType,
+} from '../../../store/api/types';
+import {
+  chooseAnimalBreedLabel,
+  chooseAnimalTypeLabel,
+} from '../../../containers/Animals/Inventory/useAnimalInventory';
+import styles from './styles.module.scss';
+
+type MenuItemProps = { iconName: IconName; text: string };
+
+const MenuItem = ({ iconName, text }: MenuItemProps) => {
+  return (
+    <div className={styles.menuItem}>
+      <Icon iconName={iconName} className={styles.menuItemIcon} />
+      {text}
+    </div>
+  );
+};
+
+type AnimalImageWithCountProps = {
+  isResponsive: boolean;
+  photoUrl?: Animal['photo_url'];
+  count?: AnimalBatch['count'];
+};
+
+const AnimalImageWithCount = ({ isResponsive, photoUrl, count }: AnimalImageWithCountProps) => {
+  return (
+    <div className={clsx(styles.animalImageWithCount, isResponsive && styles.responsive)}>
+      <div className={styles.animalImageWrapper}>
+        {photoUrl ? <img src={photoUrl} className={styles.animalImage} /> : null}
+      </div>
+      {count && <span className={styles.batchCount}>{count}</span>}
+    </div>
+  );
+};
+
+const Age = ({ birthDate, t }: { birthDate?: Animal['birth_date']; t: TFunction }) => (
+  <div className={styles.bold}>
+    {birthDate
+      ? getAge(new Date(birthDate))
+      : t('common:PROPERTY_UNKNOWN', { property: t('common:AGE') })}
+  </div>
+);
+
+const Location = ({ location, t }: { location?: string; t: TFunction }) => (
+  <div className={clsx(styles.bold, styles.location)}>
+    <Icon iconName="LOCATION" className={styles.locationIcon} />
+    <span className={styles.locationText}>{location || t('common:UNKNOWN')}</span>
+  </div>
+);
+
+type ContainerWithButtonsProps = {
+  children: ReactNode;
+  contentClassName?: string;
+  isResponsive?: boolean;
+  isEditing?: boolean;
+  options: { label: ReactNode; onClick: () => void }[];
+  onBack: () => void;
+  t: TFunction;
+};
+
+const ContainerWithButtons = ({
+  children,
+  contentClassName,
+  isResponsive,
+  isEditing,
+  options,
+  onBack,
+  t,
+}: ContainerWithButtonsProps) => {
+  return (
+    <div className={clsx(styles.containerWithButtons, isResponsive && styles.responsive)}>
+      <TextButton className={styles.backButton} onClick={onBack}>
+        <Icon iconName="CHEVRON_LEFT" className={styles.backButtonIcon} />
+      </TextButton>
+      <div className={clsx(styles.content, contentClassName)}>{children}</div>
+      <div className={styles.statusAndButton}>
+        {!isResponsive && isEditing ? <div>{t('common:EDITING')}</div> : null}
+        <ThreeDotsMenu
+          options={options}
+          classes={{ button: isEditing ? styles.editingStatusButton : '' }}
+        />
+      </div>
+    </div>
+  );
+};
+
+function isAnimal(animalOrBatch: Animal | AnimalBatch): animalOrBatch is Animal {
```

Comment:
Thank you for the suggestion! I initially thought the redundant type guard was necessary to avoid the warnings I was receiving... Also I didn't like the variable name `isAnimalType` either, so I'm happy with your solution!